frontends/web/src/components/alert/Alert.tsx

```diff
@@ -31,79 +14,54 @@ import { Button } from '../forms';
 let alertUser: (message: string, options?: AlertUserOptions) => void;
 
 type AlertUserOptions = {
-    callback?: () => void,
-    asDialog?: boolean,
+  callback?: () => void;
+  asDialog?: boolean;
 };
 
-interface State {
-    active: boolean;
-    asDialog: boolean;
-    message?: string;
-}
+const Alert = () => {
+  const [active, setActive] = useState(false);
+  const [asDialog, setAsDialog] = useState(true);
+  const [message, setMessage] = useState<string>();
+  const [callback, setCallback] = useState<AlertUserOptions['callback']>();
+  const { t } = useTranslation();
 
-class Alert extends Component<WithTranslation, State> {
-  private callback?: () => void; // Assigned when alertUser is called / Called before close.
-
-  constructor(props: WithTranslation) {
-    super(props);
-    alertUser = this.alertUser;
-    this.state = {
-      active: false,
-      asDialog: true,
-    };
-  }
-
-  private handleClose = () => {
-    if (this.callback) {
-      this.callback();
-    }
-    this.setState({
-      active: false,
-    });
-  };
-
-  private alertUser = (
-    message: string,
-    options: AlertUserOptions = {},
-  ) => {
+  alertUser = (message: string, options: AlertUserOptions = {}) => {
     const {
-      callback,
       asDialog = true,
     } = options;
-    this.callback = callback;
-    this.setState({
-      active: true,
-      asDialog,
-      message,
-    });
+    setCallback(() => options.callback);
```

Comment:
This was confusing me for a bit, i.e. I expected `setCallback(() => options.callback());` or  `setCallback(options.callback);`, but that won't work and execute the callback too early, when the alert opens instead of when the user clicks ok.

I remembered that setState also accepted a function... current react docs doesn't explain the hook well or I am blind, for future me: https://beta.reactjs.org/apis/react/useState#setstate

question as the UI doesn't render differently nor has to update (I think), does it even make sense to keep the callback in the state? (assuming there is a simpler alternative i.e. in a local variable)

Reply:
Yeah, it was also a bit confusing for me and also for a lot of people I believe, as storing a fn in react state isn't as "intuitive" as most we think. 🙂 

Yes sure, we could use local variable here and we can omit this `setCallback` thing :)